Azure/communication-ui-library-ios
# [Task] Updated Clearing Typing Indicator Logic

## Purpose
Previous implementation was using timestamps to determine when the participant should be removed from typing indicator. But after discussion with Android, we came into a conclusion that clearing indicator should be done by a redux action. And it should not be handled in view model.

## Testing
animation is from Fluent UI

https://user-images.githubusercontent.com/109105353/197626771-23953ec5-2977-4e4c-bede-37f0c74ea34f.mov


## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Because participant list is still being worked on by josh , so you need to hard code the list as the following to see typing indicator:
in participant reducer:
![image](https://user-images.githubusercontent.com/109105353/197132418-56b4b926-2e91-45b3-b394-66da4a1bd2e6.png)
2. send actions like these in Chat Service Event Handler
![image](https://user-images.githubusercontent.com/109105353/197132550-f9bae71e-e5d1-4cdf-8113-3e623b2c05d9.png)


## What to Check
1. make sure participant is removed from indicator after 8 seconds exactly 
2. make sure participant can be added to indicator once action is received
3. make sure indicator can be set to hidden if there is no more participant typing

## Other Information
